napari/layers/base/base.py

```diff
@@ -339,6 +344,20 @@ def __repr__(self):
         cls = type(self)
         return f"<{cls.__name__} layer {repr(self.name)} at {hex(id(self))}>"
 
+    def __setattr__(self, __name: str, __value: Any) -> None:
+        """Checks locks before changing value"""
+        if self.locker.is_locked(__name, hard_lock=True):
+            warnings.warn(
+                trans._(
+                    f'Attribute ({__name}) is locked is locked',
+                    deferred=True,
+                ),
+                category=UserWarning,
```

Comment:
Maybe a stacklevel for the warnign to be in the caller frame ?

Reply:
added stacklevel=2 for this one.